database/migrations/default/1750435376415_ems_person_match_auto/up.sql

```diff
@@ -0,0 +1,437 @@
+--
+CREATE
+OR REPLACE FUNCTION public.ems_update_handle_record_match_event() RETURNS trigger LANGUAGE plpgsql AS $function$
+DECLARE
+    matching_person_ids INTEGER[];
+    matching_person_record RECORD;
+BEGIN
+    raise debug '**function: ems_update_handle_record_match_event for EMS ID: % **, event: %', NEW.id, NEW._match_event_name;
+    --
+    -- Only continue if a valid _match_event_name is provided
+    --
+     IF NEW._match_event_name is null then
+        return new;
+    ELSIF NEW._match_event_name not in (
+        'match_crash_by_automation',
+        'match_person_by_manual_qa',
+        'reset_crash_match',
+        'sync_crash_pk_on_person_match',
+        'unmatch_crash_by_manual_qa',
+        'unmatch_person_by_manual_qa',
+        'update_matched_crash_ids'
+    ) then
+        RAISE EXCEPTION 'Invalid _match_event_name: `%`', NEW._match_event_name
+        USING HINT = 'Check this function definition for allowed _match_event_name values',
+            -- code 23514 is check_violation
+            ERRCODE = '23514';
+    END IF;
+    --
+    -- `unmatch_crash_by_manual_qa` event is set from the VZE using the
+    -- "Match not found" dropdown menu option
+    --
+    IF NEW._match_event_name = 'unmatch_crash_by_manual_qa' then
+        NEW.crash_match_status = 'unmatched_by_manual_qa'; 
+        NEW.crash_pk = NULL;
+        NEW.person_match_status = 'unmatched_by_manual_qa';
+        NEW.matched_person_ids = NULL;
+        NEW.person_id = NULL;
+        NEW._match_event_name = null;
+        return NEW;
+    END IF;
+    --
+    -- `unmatch_person_by_manual_qa` event is set from the VZE when
+    -- the EMS record's person ID input is cleared
+    --
+    IF NEW._match_event_name = 'unmatch_person_by_manual_qa' then
+        NEW.person_match_status = 'unmatched_by_manual_qa';
+        NEW.person_id = NULL;
+        NEW._match_event_name = null;
+        return NEW;
+    END IF;
+    --
+    -- `match_person_by_manual_qa` is sent when a person match is selected through the VZE UI
+    --
+    IF NEW._match_event_name = 'match_person_by_manual_qa' and NEW.person_id is not null then
+        -- 
+        -- Keep the record's crash_pk in sync with the provided person_id
+        --
+        SELECT id, crash_pk INTO matching_person_record 
+        FROM people_list_view 
+        WHERE id = NEW.person_id;
+
+        IF matching_person_record.crash_pk IS DISTINCT FROM NEW.crash_pk then
+            raise debug 'updating EMS record ID % crash_pk to % to match updated person_id', NEW.id, matching_person_record.crash_pk;
+            NEW.crash_pk = matching_person_record.crash_pk;
+            NEW.crash_match_status = 'matched_by_manual_qa';
+            NEW.person_match_status = 'matched_by_manual_qa';
+        END IF;
+        -- clear the _match_event_name
+        NEW._match_event_name = null;
+        return new;
+    END IF;
+    --
+    -- The update_matched_crash_ids event is set via the EMS-crash matching trigger
```

Comment:
I think this comment could be a little more descriptive. it doesnt actually update the `matched_crash_ids`, but checks the `crash_match_status` to decide whether the `reset_crash_match` event needs to happen :thinking: 

idk i just found myself being confused by what this event was doing at first

Reply:
+1 a little fuzzy on the `reset_crash_match` but after reading the `reset_crash_match` event I think its making more sense. if the crash ids have been updated, then we need to reset whatever automatic match happened because its out of date, right?

Reply:
Thanks for this feedback. I renamed this event to `handle_matched_crash_pks_updated` and expanded this comment. Let me know what you think.

Having this special event that is set from our matching trigger was kind of an aha moment for me. Prior to this change, we had to have the conditional logic in this block duplicated in a bunch of places in the crash match trigger.